### PR TITLE
Fix windows not shutting down when application stopped

### LINKED ISSUE #297: Process does not terminate when UI is closed
I just installed GRIP-0.3.0.exe on Windows 7. When I terminate the program, either by selecting File => Exit or by clicking on the red X in the upper-right corner of the window, the UI closes but the process does not terminate. I am forced to terminate the GRIP.exe process using Windows Task Manager.

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Closes #297

### CODE REVIEW

**Comment by ThomasJClark** [comment]
Cool, that was quick.  Could you explain in your PR comment what the problem turned out to be how this fixes it?

**Comment by JLLeitschuh** [comment]
I'm not sure if it fixes it. I stopped having the hanging problem.
I basically threw everything I knew at trying to fix it.
 - Making sure that the uncaught thread handler didn't recursively call itself because of an exception.
 - Making sure that the thread will die if loggers fail to log.
 - Ensuring that if all else fails, if we get an error, then we try to log and leave as soon as possible.

I'm hoping someone with a windows machine can do an install and test it.

**Comment by ThomasJClark** [comment]
ok

**Comment by JLLeitschuh** [comment]
@ThomasJClark can we merge this and to a RC2 tonight to let people see if this works on windows?

File: ui/src/main/java/edu/wpi/grip/ui/Main.java [hunk 1]
```diff
@@ -28,7 +30,9 @@
     @Inject
     private Palette palette;
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/Main.java]: I don't know why the main file has the palette.
Whatever,

**Comment by JLLeitschuh** [comment]
Can this be merged? (once I update)

**Comment by ThomasJClark** [comment]
Is there any way we can get a build tested first?  I'd rather actually trying this first, but if there's no other option I guess we can merge and do a release.

**Comment by JLLeitschuh** [comment]
I saw it on OSX but it was very, very, very rare.
Windows it happened every time.
I'm going to do a release. If you want anything else merged nows the time to do it.

**Pull request merged**